textattack/constraints/grammaticality/cola.py

```diff
@@ -0,0 +1,73 @@
+import lru
+import nltk
+from transformers import AutoModelForSequenceClassification, AutoTokenizer
+
+from textattack.constraints import Constraint
+from textattack.models.wrappers import HuggingFaceModelWrapper
+
+
+class COLA(Constraint):
+    """Constrains an attack to text that has a similar number of linguistically
+    accecptable sentences as the original text. Linguistic acceptability is
+    determined by a model pre-trained on the `CoLA dataset <https://nyu-
+    mll.github.io/CoLA/>`_. By default a BERT model is used, see the `pre-
+    trained models README <https://github.com/QData/TextAttack/tree/master/
+    textattack/models>`_ for a full list of available models or provide your
+    own model from the huggingface model hub.
+
+    Args:
+        max_diff (float): The absolute (if greater than or equal to 1) or percent (if less than 1)
+            maximum difference allowed between the number of valid sentences in the reference
+            text and the number of valid sentences in the attacked text.
+        model_name (str): The name of the pre-trained model to use for classification. The model must be in huggingface model hub.
+        compare_against_original (bool): If `True`, compare against the original text.
+            Otherwise, compare against the most recent text.
+    """
+
+    def __init__(
+        self,
+        max_diff,
+        model_name="textattack/bert-base-uncased-CoLA",
+        compare_against_original=True,
+    ):
+        super().__init__(compare_against_original)
+        if not isinstance(max_diff, float):
```

Comment:
Values like 2 is an int and not a float in Python, so we would fail this check. I think should check if `max_diff` is float or int, and base on that, determine how to calculate the threshold value for `num_valid` in `_check_constraint`. So like
``` 
if isinstance(self.max_diff, int):
      # set absolute threshold value
else:
      # set relative threshold value

# check if num_valid < threshold
```